### ISSUE TITLE
Feature request: <time> requires a unit

### ISSUE THREAD

**Issue by bfred-it** [stylelint/stylelint#3117]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Feature request, new rule

> What CSS is needed to reproduce this issue?

```css
.foo {
  animation-delay: 0;
}
```

> What did you expect to happen?

It should be an error because:

> Note: Although the number 0 is always the same regardless of unit, the unit may not be omitted. In other words, `0` is invalid and does not represent `0s` or `0ms`.
> https://developer.mozilla.org/en-US/docs/Web/CSS/time

I've been bitten by this countless times, repeatedly. 😞

**Comment by CAYdenberg** [comment]
I'm good with this. I believe `length-unit` would be sufficient as a name? `time-min-milliseconds` already lists out the declarations that would need to be targeted.

**Comment by ntwb** [comment]
@CAYdenberg The `length-*` rules, of which there is only 1 is related to _distance measurements_, not _unit lengths_, I'm thinking `time-no-missing-unit` with a single option `true`?

This should be in `stylelint-config-recommended` as it's a "Posible error" rule

----

The following patterns are considered violations:

```css
animation-delay: 0;
```

The following patterns are *not* considered violations:

```css
animation-delay: 0ms;
```

```css
animation-delay: 0s;
```

**Comment by jeddy3** [comment]
I believe this is spec validation and `animation-delay: 0;` is already covered by the [stylelint-csstree-validator](https://github.com/csstree/stylelint-validator) plugin.

I think the latest spec validation discussion is in https://github.com/stylelint/stylelint/issues/2896, including the idea of moving all the spec validation rules to a plugin pack.

**Comment by jeddy3** [comment]
Closing as covered by stylelint-csstree-validator.
